### ISSUE TITLE
X vs Y plots

### ISSUE THREAD

**Issue by LeanderK** [allegroai/trains#9]
I currently use mlflow and the functionality I use the most is the ability for quick X vs Y scatter plots. let's say I select 10 runs and want to compare them. The X vs. Y plots allow me to compare any nvalue. For example, I usually have the total number of parameters registered, then I can quickly plot the performance vs number of parameters and decide whether I should try a bigger/smaller model. Or number of iterations vs performance etc. Or choice of optimizer vs whatever. It allows me to easily explore and asses future options.

I mostly use it to compare various statistics of the model (number of parameters, layers etc.) vs various metrics I am interested in.

Having this functionality in trains would be really helpful!

(Also, how should I save information about my model. Like number of parameters, optimizer, parameters for the optimizer, number of layers etc.).

**Comment by bmartinn** [comment]
Three parts to the answer:
1. You can do probably the same thing you do in MLFlow, just add a specific value metric (single point per experiment). Then in the experiment comparison tool you will see all the experiments on the same graph. With the code sample below, if you compare two experiments, you'll get this [graph](https://github.com/bmartinn/trains/blob/dev/docs/screenshots/compare_plots.png)

```
number_layers = 10
accuracy = 0.95
Task.current_task().get_logger().report_scatter2d(
    "performance", "accuracy", iteration=0, mode='markers', scatter=[(number_layers, accuracy)])

```

2. In the experiment comparison page you can compare all the scalar values of all the experiments, that said again hyper parameters are in the "details" tab, where as metric values are in the "metric values" tab. Example [here](https://github.com/bmartinn/trains/blob/dev/docs/screenshots/compare_values.png) 

3. You can use the custom columns to quickly add specific values for comparison and then order by these columns. see example [here](https://github.com/bmartinn/trains/blob/dev/docs/screenshots/set_custom_column.png) and [here](https://github.com/bmartinn/trains/blob/dev/docs/screenshots/custom_column.png)
That said, currently you cannot add hyper-parameters as custom column, Which come to think about might be a nice feature to add. 


What do you think?

**Comment by bmartinn** [comment]
Another option for answer (1):
Create a single histogram per experiment, then when you compare experiments you get:
![Alt Text](https://github.com/bmartinn/trains/blob/dev/docs/screenshots/compare_plots_hist.png?raw=true)

Sample code:
```
number_layers = 10
accuracy = 0.95
Task.current_task().get_logger().report_vector(
    "performance", "accuracy", iteration=0, labels=['accuracy'],
    values=[accuracy], xlabels=['number_layers %d' % number_layers])
```

**Comment by bmartinn** [comment]
Regrading model information, see example here
https://github.com/allegroai/trains/blob/85f702d5941890496dd49daf3f9127c69c57dfb2/examples/manual_model_config.py#L12

Any model created following that section will contain the model configuration dictionary.

The same applies for class label enumeration:
https://github.com/allegroai/trains/blob/85f702d5941890496dd49daf3f9127c69c57dfb2/examples/manual_model_config.py#L24

Both will appear in the project page under the "*models*" tab  (not to be confused with the experiment "*model*" tab) with the configuration & labels in the corresponding tabs.

P.S.
If using keras, the model configuration will be automatically populated with the definition of the network, see example:
https://github.com/allegroai/trains/blob/master/examples/keras_tensorboard.py

Let me know if it answered your question

**Comment by bmartinn** [comment]
Closed, moved to FAQ
